Donation-platform/src/main/java/com/google/FindTrendingCharities.java

```diff
@@ -30,144 +30,152 @@
 
 public final class FindTrendingCharities {
 
-  private DbCalls db;
-  private DatastoreService ds;
-
-  // number of trending charities to be returned
-  final int MAX_NUM_OF_CHARITIES_TO_RETURN = 7;
-
-  final double CHARITY_NAV_SCALE_FACTOR = 1.25;
-
-  // should sum to 1
-  final double USER_RATING_WEIGHT = 0.75;
-  final double CHARITY_NAV_WEIGHT = 0.25;
-
-  // should sum to 1
-  final double TAGS_SCORE_WEIGHT = 0.75;
-  final double AVG_REVIEW_WEIGHT = 0.25;
-
-  private static Collection<Charity> charities;
-
-  //constructor to do set up
-  public FindTrendingCharities(DatastoreService ds) {
-    //ds = DatastoreServiceFactory.getDatastoreService();
-    this.ds = ds;
-    db = new DbCalls(ds);
-    DbSetUpUtils setUp = new DbSetUpUtils(ds, db);
-    this.charities = getAllCharities();
-    // only populate database if there is nothing in the database already
-    if (charities.size() < 1) {
-        setUp.populateDatabase();
-    }
-  }
-
-  // returns the collection of top trending charities
-  public Collection<Charity> queryDb() {
-    //Collection<Charity> charities = getAllCharities();
-    for (Charity charity : charities) {
-      double charityScore = calcCharityTrendingScore(charity);
-      charity.setTrendingScoreCharity(charityScore);
-      try {
-          db.updateCharity(charity);
-      }
-      catch (Exception e) {
-          System.out.println("unable to update charity: " + e);
-      }
+    private DbCalls db;
+    private DatastoreService ds;
+
+    // number of trending charities to be returned
+    final int MAX_NUM_OF_CHARITIES_TO_RETURN = 7;
+
+    final double CHARITY_NAV_SCALE_FACTOR = 1.25;
+
+    // should sum to 1
+    final double USER_RATING_WEIGHT = 0.75;
+    final double CHARITY_NAV_WEIGHT = 0.25;
+
+    // should sum to 1
+    final double TAGS_SCORE_WEIGHT = 0.75;
+    final double AVG_REVIEW_WEIGHT = 0.25;
+
+    private static Collection<Charity> charities;
+    private List<Charity> topTrendingCharities;
+
+    //constructor to do set up
+    public FindTrendingCharities(DatastoreService ds) {
+        //ds = DatastoreServiceFactory.getDatastoreService();
+        this.ds = ds;
+        db = new DbCalls(ds);
+        DbSetUpUtils setUp = new DbSetUpUtils(ds, db);
+        if (charities == null) {
+            this.charities = getAllCharities();
+            if (charities.size() < 1) {
+                setUp.populateDatabase();
+                this.charities = getAllCharities();
+            }
+            updateCharityScores();
```

Comment:
why are we updating scores inside findtrending.

Reply:
The large amount of functions in FindTrendingCharities are to do with updating the charity trending scores, but these functions are only called when the script is run (which makes them an offline process). Only sorting the charities is done online.

Reply:
then move it in script. I know you mentioned previously making connection to db, probably do that otherwise its very confusing

Reply:
I wasn't able to figure out how to access the db from the script. That's why I have these db-related functions here that are only called when the servlet receives a request from the script.